.buildbot.sh

```diff
@@ -2,5 +2,31 @@
 # vim: expandtab sts=0 sw=4 smarttab
 set -e
 
-echo "Test buildbot" 
+ROOT_DIR=$(pwd)/..
 
+# arg-1 : Root directory
+# arg-2 : cmake toolchain file
+function build_boehmGC()
+{
+    ARCH=$(echo ${2} | cut -d '.' -f 1)
+    SRC_DIR=${1}/build # build/source directory name used for all repos
+    BUILD_DIR=${1}/bdwgc-build
+    INSTALL_DIR=${1}/install
+    BUILD_OPTS="-DCMAKE_BUILD_TYPE=Debug \
+                -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} \
+                -Denable_gcj_support=OFF \
+                -Denable_parallel_mark=OFF \
+                -Denable_threads=OFF \
+                -Denable_dynamic_loading=OFF"
+
+
+    mkdir -p ${BUILD_DIR}
+    cmake -B ${BUILD_DIR} -S ${SRC_DIR} ${BUILD_OPTS} -DCMAKE_TOOLCHAIN_FILE=${2}
```

Comment:
I think this should be:

```
mkdir -p build
cmake -B build -S . ${BUILD_OPTS} -DCMAKE_TOOLCHAIN_FILE=${2}
```

This means that we don't end up writing files *above* the repository's directory, which seems safer (we don't know what the parent directory above this repository contains AFAIK -- but I might be wrong!).

Reply:
I agree that we should not normally be writing to parent directories at all.  However, CMake discourages in-source-tree building generally.

Reply:
It's common with `cmake` to write to a non-git-tracked directory *inside* the repository though -- you don't have to write to a parent directory.